src/modules/EventModule.cpp

```diff
@@ -0,0 +1,463 @@
+						return;
+					}
+					if (time > 24)
+					{
+						UmikoBot::Instance().createMessage(message.channelId(), "**Currently you can't launch events with an expiry time of more than one day**.");
+						return;
+					}
+					if (config.eventTimer != nullptr) //Delete the previous timer
+					{
+						delete config.eventTimer;
+						config.eventTimer = nullptr;
+					}
+					config.eventTimer = new QTimer;
+
+					config.eventTimer->setInterval(time * 3600000);
+					config.eventTimer->setSingleShot(true);
+
+					if (args.at(1) == "HighRiskHighReward")
+					{
+						QString num = QString::number(time);
+
+						Discord::Embed embed;
+						embed.setColor(15844367);
+						embed.setTitle("**HighRiskHighReward** event has been launched for `" + num + "` hour(s)!");
+						embed.setDescription("Give command `!event` to see what event is running and its changes!");
+
+						config.isEventRunning = true;
+						config.eventHighRiskHighRewardRunning = true;
+						currencyConfig.stealSuccessChance -= highRiskRewardStealDecrease;
+						UmikoBot::Instance().createMessage(message.channelId(), embed);
+					}
+					if (args.at(1) == "LowRiskLowReward")
+					{
+						QString num = QString::number(time);
+
+						Discord::Embed embed;
+						embed.setColor(15844367);
+						embed.setTitle("**LowRiskLowReward** event has been launched for `" + num + "` hour(s)!");
+						embed.setDescription("Give command `!event` to see what event is running and its changes!");
+
+						config.isEventRunning = true;
+						config.eventLowRiskLowRewardRunning = true;
+						currencyConfig.stealSuccessChance += lowRiskRewardStealIncrease;
+						UmikoBot::Instance().createMessage(message.channelId(), embed);
+					}
+					auto guildID = channel.guildId();
+					auto chan = message.channelId();
+
+					QObject::connect(config.eventTimer, &QTimer::timeout, [this, guildID, chan]()
+						{
+							CurrencyModule* currencyModule = static_cast<CurrencyModule*>(UmikoBot::Instance().GetModuleByName("currency"));
+							auto& currencyConfig = currencyModule->getServerData(guildID);
+
+							auto& config = getEventServerData(guildID);
+							if (config.eventHighRiskHighRewardRunning)
+							{
+								Discord::Embed embed;
+								embed.setColor(15158332);
+								embed.setTitle("**HighRiskHighReward** event has **ended**!");
+
+								config.isEventRunning = false;
+								config.eventHighRiskHighRewardRunning = false;
+								currencyConfig.stealSuccessChance += highRiskRewardStealDecrease;
+								UmikoBot::Instance().createMessage(chan, embed);
+							}
+							else if (config.eventLowRiskLowRewardRunning)
+							{
+								Discord::Embed embed;
+								embed.setColor(15158332);
+								embed.setTitle("**LowRiskLowReward** event has **ended**!");
+
+								config.isEventRunning = false;
+								config.eventLowRiskLowRewardRunning = false;
+								currencyConfig.stealSuccessChance -= lowRiskRewardStealIncrease;
```

Comment:
You shouldn't modify the actual steal success chance variable, as that is saved every few minutes (and it represents the normal steal success chance).

Reply:
I restore the real steal success chance every time the bot starts (after crash)!

Reply:
I still feel like it'd be safer if we just didn't modify those, let's see what @Gaztin says

Reply:
Can we make a copy of the real `stealSuccessChance` and store that in a variable and tweak the copied variable?

Reply:
Yeah don't modify those. What you can do is change how we calculate the success chance when a user actually performs a steal command, and use the event system there 😊

Reply:
Okay 😄